### ISSUE TITLE
Add OpenShift support for our release pipelines

### ISSUE THREAD

**Issue by a-roberts** [tektoncd/dashboard#1310]
# Expected behavior
I should be able to run the release pipeline on OpenShift 4.x

# Actual behavior
It's only known to work on Docker Desktop and (very probably) Google Cloud

# Steps to reproduce the problem

1. Run the release pipeline on OpenShift
2. See
```
[build : build-static] W: chmod 0700 of directory /var/lib/apt/lists/partial failed - SetupAPTPartialDirectory (1: Operation not permitted)
[build : build-static] E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
[build : build-static] E: Unable to lock directory /var/lib/apt/lists/
```
from the PipelineRun

# Environment
- Kubernetes Platform:
OpenShift

- Tekton Pipelines version:
Any, but i'm using 0.11.2

- Tekton Triggers version:

- Tekton Dashboard version:
Any

# Additional Info

Likely need
```
oc adm policy add-scc-to-user anyuid -z default -n tekton-pipelines
```
and to update our docs.

May as well do this, I think @AlanGreene uses CRC and it's useful to have the flexibility of environments. It's also useful as part of https://github.com/tektoncd/pipeline/issues/2417 (I want to see the impact on memory for a platform other than Docker Desktop personally).

**Comment by a-roberts** [comment]
Almost there

![image](https://user-images.githubusercontent.com/12814972/79861267-08a7c080-83cc-11ea-8b4e-afb8b88e899b.png)

build step completes now, we're using the Docker socket though

```
  Warning  FailedMount  97s                   kubelet, worker4.gulp.os.fyre.ibm.com  Unable to attach or mount volumes: unmounted volumes=[docker-socket], unattached volumes=[gcp-secret tekton-internal-workspace tekton-internal-results default-token-gpfsn tekton-internal-scripts tekton-internal-home tekton-internal-tools tekton-internal-downward dashboard-release-run-xf4tj-pvc docker-socket tekton-internal-secret-volume-default-dockercfg-ncb8b-6smz4]: timed out waiting for the condition
```

so suppose I'll have to switch it out for buildah (and hope that works a treat on Docker Desktop, or we have another folder for OpenShift resources).

I'll pause this while I respin 0.6.1 now to pick up some fixes.

**Comment by a-roberts** [comment]
Update on this one, pushed the latest progress to my branch, see https://github.com/tektoncd/dashboard/compare/master...a-roberts:openshiftrelease?expand=1

The lcokdown.py script we use makes use of a Docker client via the Docker socket, so this is going to be a problem on OpenShift. Perhaps I can use buildah instead to list the images, but for now I've omitted this step just to see what happens and how much progress I get with `ko`. I've also added to. the documentation which features an example ServiceAccount and namespace that's isolated from everything else - and that likely needs to be augmented with RBAC for use with `tkn`.
